src/storage/store/shard.rs

```
@@ -208,6 +228,20 @@ impl ShardStore {
         }
     }
 
+    pub fn min_block_number(&self) -> Result<u64, ShardStorageError> {
+        let first_shard_chunk = get_first_or_last_shard_chunk(&self.db, FirstOrLast::First)?;
+        match first_shard_chunk {
+            None => Ok(0),
```

still unsure about returning 0/1 here. maybe this should be handled at the callsite?

---

for now I put a band-aid on the callsite as max(1)

---

Hmm, I think it's more correct to return and Error if there are no blocks instead of having to do max(1). And and have the caller decide if it should propagate the error or return 0.